pkg/image/validation.go

```diff
@@ -0,0 +1,168 @@
+package image
+
+import (
+	"fmt"
+	"strings"
+)
+
+func ValidateDefinition(definition *Definition) error {
+	err := validateImage(definition)
+	if err != nil {
+		return fmt.Errorf("error validating image: %w", err)
+	}
+	err = validateOperatingSystem(definition)
+	if err != nil {
+		return fmt.Errorf("error validating operating system: %w", err)
+	}
+
+	return nil
+}
+
+func validateImage(definition *Definition) error {
+	if definition.Image == (Image{}) {
+		return fmt.Errorf("image not defined")
+	}
+	if definition.Image.ImageType == "" {
+		return fmt.Errorf("imageType not defined")
+	} else if definition.Image.ImageType != "iso" && definition.Image.ImageType != "raw" {
+		return fmt.Errorf("invalid imageType, should be 'iso' or 'raw'")
+	}
+	if definition.Image.BaseImage == "" {
+		return fmt.Errorf("baseImage not defined")
+	}
+	if definition.Image.OutputImageName == "" {
+		return fmt.Errorf("outputImageName not defined")
+	}
+	return nil
+}
+
+func validateOperatingSystem(definition *Definition) error {
+	if checkIfOperatingSystemDefined(&definition.OperatingSystem) {
+		return nil
+	}
+	err := validateKernelArgs(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating kernal args: %w", err)
+	}
+	err = validateSystemd(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating systemd args: %w", err)
+	}
+	err = validateUsers(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating users: %w", err)
+	}
+	err = validateSuma(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating suma: %w", err)
+	}
+	return nil
+}
+
+func checkIfOperatingSystemDefined(os *OperatingSystem) bool {
+	return len(os.KernelArgs) == 0 &&
+		len(os.Users) == 0 &&
+		len(os.Systemd.Enable) == 0 && len(os.Systemd.Disable) == 0 &&
+		os.Suma == (Suma{})
+}
+
+func validateKernelArgs(os *OperatingSystem) error {
+	seenKeys := make(map[string]bool)
+
+	for _, arg := range os.KernelArgs {
+		if !strings.Contains(arg, "=") {
+			if _, exists := seenKeys[arg]; exists {
+				return fmt.Errorf("duplicate kernel arg found: '%s'", arg)
+			}
+			seenKeys[arg] = true
+		} else {
+			parts := strings.SplitN(arg, "=", 2)
+			if len(parts) != 2 {
+				return fmt.Errorf("invalid kernel arg: '%s', expected format key=value or a single string", arg)
+			}
+			key, value := parts[0], parts[1]
+			if key == "" {
+				return fmt.Errorf("kernel arg value '%s' has no key but has '='", value)
+			}
+			if value == "" {
+				return fmt.Errorf("kernel arg '%s' has no value", key)
+			}
+
+			if _, exists := seenKeys[key]; exists {
+				return fmt.Errorf("duplicate kernel arg found: '%s'", key)
+			}
+			seenKeys[key] = true
+		}
```

Comment:
It feels like this can be simplified quite a lot:
```
key := arg

parts := strings.SplitN(arg, "=", 2)
if len(parts) == 2 {
    var value string
    key, value = parts[0], parts[1]
    if key == "" {
	return fmt.Errorf("kernel arg value '%s' has no key", value)
    }
    if value == "" {
	return fmt.Errorf("kernel arg '%s' has no value", key)
    }
}

if _, exists := seenKeys[key]; exists {
    return fmt.Errorf("duplicate kernel arg found: '%s'", key)
}
seenKeys[key] = true
```